test/cycle.js

```diff
@@ -134,4 +139,115 @@ describe('Cycle', function () {
       }, 10);
     });
   });
+
+  describe('isolate()', function () {
+    it('should throw if first argument is not a function', function () {
+      assert.throws(() => {
+        Cycle.isolate('not a function');
+      }, /First argument given to Cycle\.isolate\(\) must be a 'dialogue' function/i);
+    });
+
+    it('should throw if second argument is not a string', function () {
+      function MyDialogue() {}
+      assert.throws(() => {
+        Cycle.isolate(MyDialogue, null);
+      }, /Second argument given to Cycle\.isolate\(\) must be a string for 'scope'/i);
+    });
+
+    it('should return a function', function () {
+      function MyDialogue() {}
+      const scopedMyDialogue = Cycle.isolate(MyDialogue, `myScope`);
+      assert.strictEqual(typeof scopedMyDialogue, `function`);
+    });
+
+    it('should make a new scope if second argument is undefined', function () {
+      function MyDialogue() {}
+      const scopedMyDialogue = Cycle.isolate(MyDialogue);
+      assert.strictEqual(typeof scopedMyDialogue, `function`);
+    });
+
+    describe('scopedDialogue', function () {
+      it('should return scoped sinks', function () {
+        function driver() {
+          return {};
+        }
+
+        function MyDialogue(sources) {
+          return {
+            other: Rx.Observable.just('a')
+          };
+        }
+        const scopedMyDialogue = Cycle.isolate(MyDialogue);
+        const scopedSinks = scopedMyDialogue({other: driver()});
+
+        assert.strictEqual(typeof scopedSinks, `object`);
+        scopedSinks.other.subscribe(x => {
+          assert.strictEqual(x, 'a');
+        })
+      })
+
+      it('should isolate sources', function () {
```

Comment:
This test is actually testing **sink** isolation, not **source** isolation. But to keep it simple, since Cycle DOM and Cycle HTTP Driver already have tests for isolateSource/isolateSink, you can just write tests here that assert that isolateSource and isolateSink were **called** when calling `isolate()`. We don't need to assert that actual scoping isolation occurs, because that's a per-driver feature.